calico_kubernetes/calico_kubernetes.py

```diff
@@ -269,31 +270,64 @@ def _get_api_token(self):
         auth_data = json.loads(json_string)
         return auth_data['BearerToken']
 
-    def _generate_rules(self):
+    def _generate_rules(self, pod):
         """
-        Generate the Profile rules that have been specified on the Pod's ports.
+        Generate Rules takes human readable policy strings in annotations
+        and creates argument arrays for calicoctl
 
-        We only create a Rule for a port if it has 'allowFrom' specified.
+        :return two lists of rules(arg lists): inbound list of rules (arg lists)
+        outbound list of rules (arg lists)
+        """
 
-        The Rule is structured to match the Calico etcd format.
+        namespace, ns_tag = self._get_namespace_and_tag(pod)
+
+        # kube-system services need to be accessed by all namespaces
+        if namespace == "kube-system" :
+            print "using kube-system, allow all"
+            return [["allow"]], [["allow"]]
 
-        :return list() rules: the rules to be added to the Profile.
-        """
         inbound_rules = [
-            {
-                'action': 'allow',
-            },
+            ["allow", "from", "tag", ns_tag]
         ]
 
         outbound_rules = [
-            {
-                'action': 'allow',
-            },
+            ["allow"]
         ]
+
+        print("Getting Policy Rules from Annotation of pod %s" % pod)
+
+        annotations = self._get_metadata(pod, "annotations")
+
+        # Find policy block of annotations
+        if annotations and "policy" in annotations.keys():
+            # Remove Default Rule (Allow Namespace)
+            inbound_rules = []
+            rules = annotations["policy"]
+
+            # Rules separated by semicolons
+            for rule in rules.split(";"):
+                args = rule.split(" ")
+
+                # Replace labels with tags
+                # key=value -> namespace_key_value
+                # TODO: account for improper formatting
```

Comment:
What is this TODO referring to?

In general, we don't want TODOs left around in the code.  If this is something you think needs to be done for this PR, lets talk.  Otherwise, stick in on the backlog somewhere we can remember it and we'll get to it later.